src/main/java/nextstep/subway/domain/line/LineStation.java

```diff
@@ -0,0 +1,144 @@
+    private Station preStation;
+
+    @Column(nullable = false)
+    private int distance;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "line_id", foreignKey = @ForeignKey(name = "fk_line_station_to_line"))
+    private Line line;
+
+    public LineStation() {
+    }
+
+    public LineStation(Station station, Station preStation, int distance, Line line) {
+        this.station = station;
+        this.preStation = preStation;
+        this.distance = distance;
+        this.line = line;
+    }
+
+    public StationStatus checkStationStatus(Station station) {
+        if (station.equals(this.station)) {
+            return new StationStatus(station, this, StationPosition.DOWNSTATION);
+        }
+        if (station.equals(this.preStation)) {
+            return new StationStatus(station, this, StationPosition.UPSTATION);
+        }
+        return null;
+    }
+
+    public int distanceCompare(int distance) {
+        return Integer.compare(this.distance, distance);
+    }
+
+    public void splitLineStation(StationPosition stationPosition, Station interStation, int subtractDistance) {
+        if (stationPosition == StationPosition.UPSTATION) {
+            preStation = interStation;
+        }
+        if (stationPosition == StationPosition.DOWNSTATION) {
+            station = interStation;
+        }
+        distance -= subtractDistance;
+    }
+
+    public void validate(LineStations lineStations) {
+        StationRegisterStatus upStationStatus = lineStations.getStationRegisterStatus(preStation);
+        StationRegisterStatus downStationStatus = lineStations.getStationRegisterStatus(station);
+        checkRegisteredStationExist(upStationStatus, downStationStatus);
+        checkBothStationRegistered(upStationStatus, downStationStatus);
+        upStationStatus.validate(StationPosition.UPSTATION, distance, station);
+        downStationStatus.validate(StationPosition.DOWNSTATION, distance, preStation);
+    }
+
+    private void checkRegisteredStationExist(StationRegisterStatus upStationStatus,
+            StationRegisterStatus downStationStatus) {
+        if (upStationStatus.isEmpty() && downStationStatus.isEmpty()) {
+            throw new IllegalArgumentException(ErrorMessage.BOTH_STATIONS_NOT_REGISTERED);
+        }
+    }
+
+    private void checkBothStationRegistered(StationRegisterStatus upStationStatus,
+            StationRegisterStatus downStationStatus) {
+        if (!upStationStatus.isEmpty() && !downStationStatus.isEmpty()) {
+            throw new IllegalArgumentException(ErrorMessage.ALREADY_REGISTERED_SECTION);
+        }
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public Station getStation() {
+        return station;
+    }
+
+    public Station getPreStation() {
+        return preStation;
+    }
+
+    public int getDistance() {
+        return distance;
+    }
+
+    public Line getLine() {
+        return line;
+    }
+
+    public void setLine(Line line) {
```

Comment:
```suggestion
    public void addLine(Line line) {
```

개인적으로는, 객체지향 관점에서 말하는 명확한 메시지를 전달해준다의 사고를 훈련하기 위해서
`set_`, `get_` 으로 시작하는 네이밍을 왠만하면 사용안할려고 하는 편이에요.
(물론, 무조건 사용안할수는 없겠지만요 ^^;)

Reply:
dto에서 사용하는 getter, setter 외에는 목적에 따라 메소드명을 작성하는 습관을 길러봐야겠네요! 😵‍💫